From: VLCDotNet CI <ci@vlcdotnet>
Subject: [PATCH] contrib: prefer upstream package sources over VideoLAN mirror

Current GitHub-hosted builds can reach a number of upstream project archives but
repeatedly time out against `downloads.videolan.org`, even after switching the
mirror root from HTTP to HTTPS. After disabling the unused optional `a52` and
`dca` contribs at the build-script level, the remaining fetch blockers are the
packages that still rely on VideoLAN-only or mirror-first download rules.

Prefer the original package URL before the VideoLAN contrib mirror, then reroute
the remaining required VideoLAN-only packages to stable upstream HTTPS archives.
---
 contrib/src/dav1d/rules.mak  | 10 +++++-----
 contrib/src/dvbpsi/rules.mak |  8 ++++----
 contrib/src/main.mak         |  4 ++--
 3 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/contrib/src/dav1d/rules.mak b/contrib/src/dav1d/rules.mak
index eed3d28..675d2a1 100644
--- a/contrib/src/dav1d/rules.mak
+++ b/contrib/src/dav1d/rules.mak
@@ -1,7 +1,7 @@
 # libdav1d
 
 DAV1D_VERSION := 1.5.1
-DAV1D_URL := $(VIDEOLAN)/dav1d/$(DAV1D_VERSION)/dav1d-$(DAV1D_VERSION).tar.xz
+DAV1D_URL := https://code.videolan.org/videolan/dav1d/-/archive/$(DAV1D_VERSION)/dav1d-$(DAV1D_VERSION).tar.gz
 
 PKGS += dav1d
 ifeq ($(call need_pkg,"dav1d"),)
@@ -10,13 +10,13 @@ endif
 
 DAV1D_CONF = -D enable_tests=false -D enable_tools=false
 
-$(TARBALLS)/dav1d-$(DAV1D_VERSION).tar.xz:
-	$(call download_pkg,$(DAV1D_URL),dav1d)
+$(TARBALLS)/dav1d-$(DAV1D_VERSION).tar.gz:
+	$(call download,$(DAV1D_URL))
 #	$(call download_git,$(DAV1D_GITURL),,$(DAV1D_HASH))
 
-.sum-dav1d: dav1d-$(DAV1D_VERSION).tar.xz
+.sum-dav1d: dav1d-$(DAV1D_VERSION).tar.gz
 
-dav1d: dav1d-$(DAV1D_VERSION).tar.xz .sum-dav1d
+dav1d: dav1d-$(DAV1D_VERSION).tar.gz .sum-dav1d
 	$(UNPACK)
 	$(MOVE)
 
diff --git a/contrib/src/dvbpsi/rules.mak b/contrib/src/dvbpsi/rules.mak
index 12efdd3..12a33e6 100644
--- a/contrib/src/dvbpsi/rules.mak
+++ b/contrib/src/dvbpsi/rules.mak
@@ -1,19 +1,19 @@
 # dvbpsi
 
 DVBPSI_VERSION := 1.3.3
-DVBPSI_URL := $(VIDEOLAN)/libdvbpsi/$(DVBPSI_VERSION)/libdvbpsi-$(DVBPSI_VERSION).tar.bz2
+DVBPSI_URL := https://code.videolan.org/videolan/libdvbpsi/-/archive/$(DVBPSI_VERSION)/libdvbpsi-$(DVBPSI_VERSION).tar.gz
 
 PKGS += dvbpsi
 ifeq ($(call need_pkg,"libdvbpsi >= 1.2.0"),)
 PKGS_FOUND += dvbpsi
 endif
 
-$(TARBALLS)/libdvbpsi-$(DVBPSI_VERSION).tar.bz2:
+$(TARBALLS)/libdvbpsi-$(DVBPSI_VERSION).tar.gz:
 	$(call download,$(DVBPSI_URL))
 
-.sum-dvbpsi: libdvbpsi-$(DVBPSI_VERSION).tar.bz2
+.sum-dvbpsi: libdvbpsi-$(DVBPSI_VERSION).tar.gz
 
-libdvbpsi: libdvbpsi-$(DVBPSI_VERSION).tar.bz2 .sum-dvbpsi
+libdvbpsi: libdvbpsi-$(DVBPSI_VERSION).tar.gz .sum-dvbpsi
 	$(UNPACK)
 	$(UPDATE_AUTOCONFIG) && cd $(UNPACK_DIR) && mv config.guess config.sub .auto
 	$(APPLY) $(SRC)/dvbpsi/dvbpsi-noexamples.patch
diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index a4a930d..e7d50f3 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -269,8 +269,8 @@ else
 download = $(error Neither curl nor wget found!)
 endif
 
-download_pkg = $(call download,$(CONTRIB_VIDEOLAN)/$(2)/$(lastword $(subst /, ,$(@)))) || \
-	( $(call download,$(1)) && echo "Please upload this package $(lastword $(subst /, ,$(@))) to our FTP" )
+download_pkg = $(call download,$(1)) || \
+	$(call download,$(CONTRIB_VIDEOLAN)/$(2)/$(lastword $(subst /, ,$(@))))
 
 ifeq ($(shell which xz >/dev/null 2>&1 || echo FAIL),)
 XZ = xz
